src/cuda/cuda-pointcloud.cu

```diff
@@ -17,15 +17,40 @@ void deproject_pixel_to_point_cuda(float points[3], const struct rs2_intrinsics
     //assert(intrin->model != RS2_DISTORTION_BROWN_CONRADY); // Cannot deproject to an brown conrady model
     float x = (pixel[0] - intrin->ppx) / intrin->fx;
     float y = (pixel[1] - intrin->ppy) / intrin->fy;    
-    if(intrin->model == RS2_DISTORTION_INVERSE_BROWN_CONRADY)
+
+    float xo = x;
+    float yo = y;
+
+    if (intrin->model == RS2_DISTORTION_INVERSE_BROWN_CONRADY)
     {
-        float r2  = x*x + y*y;
-        float f = 1 + intrin->coeffs[0]*r2 + intrin->coeffs[1]*r2*r2 + intrin->coeffs[4]*r2*r2*r2;
-        float ux = x*f + 2*intrin->coeffs[2]*x*y + intrin->coeffs[3]*(r2 + 2*x*x);
-        float uy = y*f + 2*intrin->coeffs[3]*x*y + intrin->coeffs[2]*(r2 + 2*y*y);
-        x = ux;
-        y = uy;
-    } 
+        // need to loop until convergence 
+        // 10 iterations determined empirically
+        for (int i = 0; i < 10; i++)
+        {
+            float r2 = x * x + y * y;
+            float icdist = (float)1 / (float)(1 + ((intrin->coeffs[4] * r2 + intrin->coeffs[1])*r2 + intrin->coeffs[0])*r2);
+            float xq = x / icdist;
+            float yq = y / icdist;
+            float delta_x = 2 * intrin->coeffs[2] * xq*yq + intrin->coeffs[3] * (r2 + 2 * xq*xq);
+            float delta_y = 2 * intrin->coeffs[3] * xq*yq + intrin->coeffs[2] * (r2 + 2 * yq*yq);
+            x = (xo - delta_x)*icdist;
+            y = (yo - delta_y)*icdist;
+        }
+    }
+    if (intrin->model == RS2_DISTORTION_BROWN_CONRADY)
```

Comment:
else

Reply:
Other distortion models are not supported currently on CUDA, so we have nothing to do on else , it will behave like  none distortion.

Reply:
else if